catroid/src/test/java/org/catrobat/catroid/test/content/actions/IfOnEdgeBounceActionTest.kt

```diff
@@ -0,0 +1,123 @@
+                    getExpectedDirection(it, (-179..-1).toSet(), 0)),
+            arrayOf("RIGHT_BOUNCE", RIGHT_BORDER_POSITION, 0f, BOUNCE_RIGHT_POSITION, 0f, it,
+                    getExpectedDirection(it, (0..179).toSet(), 0)),
+            arrayOf("NO_BOUNCE", 0f, 0f, 0f, 0f, it, it.toFloat())
+        ) }.plus(listOf(
+            arrayOf("LEFT_TOP_BOUNCE", LEFT_BORDER_POSITION, TOP_BORDER_POSITION,
+            BOUNCE_LEFT_POSITION, BOUNCE_TOP_POSITION, -45, 135f),
+            arrayOf("RIGHT_TOP_NO_BOUNCE", RIGHT_BORDER_POSITION, TOP_BORDER_POSITION,
+                    BOUNCE_RIGHT_POSITION, BOUNCE_TOP_POSITION, -135, -135f),
+            arrayOf("RIGHT_TOP_BOUNCE", RIGHT_BORDER_POSITION, TOP_BORDER_POSITION,
+                    BOUNCE_RIGHT_POSITION, BOUNCE_TOP_POSITION, 45, -135f),
+            arrayOf("LEFT_BOTTOM_BOUNCE", LEFT_BORDER_POSITION, BOTTOM_BORDER_POSITION,
+                    BOUNCE_LEFT_POSITION, BOUNCE_BOTTOM_POSITION, -135, 45f),
+            arrayOf("LEFT_BOTTOM_NO_BOUNCE", LEFT_BORDER_POSITION, BOTTOM_BORDER_POSITION,
+                    BOUNCE_LEFT_POSITION, BOUNCE_BOTTOM_POSITION, 45, 45f),
+            arrayOf("RIGHT_BOTTOM_NO_BOUNCE", RIGHT_BORDER_POSITION, BOTTOM_BORDER_POSITION,
+                    BOUNCE_RIGHT_POSITION, BOUNCE_BOTTOM_POSITION, -45, -45f),
+            arrayOf("RIGHT_BOTTOM_BOUNCE", RIGHT_BORDER_POSITION, BOTTOM_BORDER_POSITION,
+                    BOUNCE_RIGHT_POSITION, BOUNCE_BOTTOM_POSITION, 135, -45f)
+        ))
+
+        private const val WIDTH = 100f
+        private const val HEIGHT = 100f
+        private const val SCREEN_WIDTH = 480
+        private const val SCREEN_HEIGHT = 800
+        private const val TOP_BORDER_POSITION = SCREEN_HEIGHT / 2f
+        private const val BOTTOM_BORDER_POSITION = -TOP_BORDER_POSITION
+        private const val RIGHT_BORDER_POSITION = SCREEN_WIDTH / 2f
+        private const val LEFT_BORDER_POSITION = -RIGHT_BORDER_POSITION
+        private const val BOUNCE_TOP_POSITION = TOP_BORDER_POSITION - HEIGHT / 2f
+        private const val BOUNCE_BOTTOM_POSITION = -BOUNCE_TOP_POSITION
+        private const val BOUNCE_RIGHT_POSITION = RIGHT_BORDER_POSITION - WIDTH / 2f
+        private const val BOUNCE_LEFT_POSITION = -BOUNCE_RIGHT_POSITION
+
+        private fun getExpectedDirection(initialDirection: Int, bounceRange: Set<Int>, axis: Int): Float {
+                return if (initialDirection in bounceRange) {
+                    (if (initialDirection >= 0) axis else -axis) - initialDirection
+                } else {
+                    initialDirection
+                }.toFloat()
+        }
+    }
+
+    @Before
+    fun setUp() {
+        sprite = SingleSprite("Test")
+        sprite.look.width = WIDTH
+        sprite.look.height = HEIGHT
+        sprite.look.setPositionInUserInterfaceDimensionUnit(0f, 0f)
+        project = Project(MockUtil.mockContextForProject(), "Test", false)
```

Comment:
The `project` variable could be local.
But you can completely avoid it, again with scope functions:
```
val context = MockUtil.mockContextForProject()
ProjectManager.getInstance().currentProject = Project(context, "Test", false).apply {
            xmlHeader.virtualScreenWidth = SCREEN_WIDTH
            xmlHeader.virtualScreenHeight = SCREEN_HEIGHT
}
```
OR 
```
Project(MockUtil.mockContextForProject(), "Test", false).apply {
            xmlHeader.virtualScreenWidth = SCREEN_WIDTH
            xmlHeader.virtualScreenHeight = SCREEN_HEIGHT
            ProjectManager.getInstance().currentProject = this
}
```

Reply:
But maybe you can find a better way here: https://kotlinlang.org/docs/reference/scope-functions.html